src/java/org/apache/cassandra/db/compaction/CompactionTask.java

```diff
@@ -301,7 +301,7 @@ public boolean apply(SSTableReader sstable)
                     else if (e instanceof CompactionInterruptedException)
                         throw (CompactionInterruptedException) e;
                     else
-                        throw new IllegalStateException(e);
+                        throw new RuntimeException(e);
```

Comment:
We don't need to catch these exceptions here. `WrappedRunnable.run()` will do it.